.github/workflows/qat.yml

```diff
@@ -69,21 +84,45 @@ jobs:
               hash=`curl --no-progress-meter --insecure $hash_url`
               sleep 5
           done
+    outputs:
+      status: ${{ steps.check_files.outputs.files_exists }}
 
-      - name: Install Playwright
-        if: ${{ env.RUN_QAT == 'true' }}
-        run: |
-          . "$NVM_DIR/nvm.sh"
-
-          nvm install
+  checkly-deploy:
+    name: Checkly Run and Deploy
+    runs-on: ubuntu-latest
+    timeout-minutes: 90
+    # if the check_files step in the check-qat job doesn't fail
+    needs:
+      - check-qat
+    if: needs.check-qat.outputs.status == 'true'
 
-          # Playwright needs to be added in package.json as dependency
-          npm ci
+    concurrency:
+      group: ${{ github.ref_name || github.run_id }}-checkly-qat
+      cancel-in-progress: true
 
-      - name: Run Playwright tests
-        if: ${{ env.RUN_QAT == 'true' }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0 # Check out the full git history
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16 # Checkly CLI requires Node.js 16 or higher.
+      - name: Restore or cache node_modules # Restore node_modules cache if available
+        id: cache-node-modules
+        uses: actions/cache@v3
+        with:
+          path: node_modules
+          key: node-modules-${{ hashFiles('package-lock.json') }}
+      - name: Install dependencies # Install NPM dependencies if cache miss
+        if: steps.cache-node-modules.outputs.cache-hit != 'true'
+        run: npm ci
+      - name: Run checks # Run Checkly checks and record test session
+        id: run-checks
         run: |
-          if [ "${{ inputs.QA_SITE_DOMAIN }}" != '' ]; then
-            export ENVIRONMENT_URL="https://${{ env.FEATURE_BRANCH_NAME }}.${{ inputs.QA_SITE_DOMAIN }}"
-          fi
-          npm run ${{ env.NPM_QAT_SCRIPT }}
+          npx checkly test --config=$(find . -name checkly.config.ts) -e ENVIRONMENT_URL=https://${{ env.FEATURE_BRANCH_NAME }}.${{ inputs.QA_SITE_DOMAIN }} -e CHECKLY_REPO_SHA=${{ env.CHECKLY_REPO_SHA }} --reporter=github --record
+          cat checkly-github-report.md > $GITHUB_STEP_SUMMARY
+      - name: Deploy checks # Deploy checks to Checkly for debugging
+        # Only run if the pull request is merging into the main branch
+        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }}
```

Comment:
We may not need to check the event name, just the branch being main. I.e. the only events which are allowed on main are approved PR merges.

Reply:
I don't think we need to because of the branch protection on main preventing a pull request event on there but I thinks we can keep this in for completeness.